src/Source/DoctrineORMQuerySourceIterator.php

```diff
@@ -135,30 +170,38 @@ public function rewind()
      */
     public function setDateTimeFormat($dateTimeFormat)
     {
-        $this->dateTimeFormat = $dateTimeFormat;
-    }
+        @trigger_error('Passing a dateTimeFormat string is deprecated since 1.7.2, to be removed in 2.0. '.
+            'The source should be initialized in the constructor with an array of DataFormatterInterfaces instead',
+            E_USER_DEPRECATED);
 
-    /**
-     * @return string
-     */
-    public function getDateTimeFormat()
-    {
-        return $this->dateTimeFormat;
+        $newFormatter = new DateTimeFormatter($dateTimeFormat);
+        $this->formatters[$newFormatter->getPriority()] = $newFormatter;
     }
 
     /**
-     * @param $value
+     * @param mixed        $target
+     * @param PropertyPath $propertyPath
      *
-     * @return null|string
+     * @return mixed
      */
-    protected function getValue($value)
+    protected function getValue($target)
     {
-        if (is_array($value) || $value instanceof \Traversable) {
-            $value = null;
-        } elseif ($value instanceof \DateTime) {
-            $value = $value->format($this->dateTimeFormat);
-        } elseif (is_object($value)) {
-            $value = (string) $value;
+        if (func_num_args() == 2 && func_get_arg(1) instanceof PropertyPath) {
```

Comment:
Please add `NEXT_MAJOR` instructions where needed, as described in CONTRIBUTING.md